From 48921e9febe0ead98057086b9fba2307f032df0f Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Sat, 28 Mar 2015 14:37:34 +0800
Subject: [PATCH] HDMI: remove limited conditions of calling switch_set_state. 
      Application should read /sys/class/display/HDMI/audioinfo       to
 detect sink device support audio or not.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-core.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
index 4fecefed0c65..290462d93267 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
@@ -217,10 +217,7 @@ static void hdmi_wq_insert(struct hdmi *hdmi)
 		/*hdmi->autoset = 0;*/
 		hdmi_wq_set_video(hdmi);
 		#ifdef CONFIG_SWITCH
-		if ((hdmi->edid.baseaudio_support &&
-		     hdmi->edid.sink_hdmi) ||
-		    rk_fb_get_display_policy() == DISPLAY_POLICY_BOX)
-			switch_set_state(&(hdmi->switchdev), 1);
+		switch_set_state(&(hdmi->switchdev), 1);
 		#endif
 		hdmi_wq_set_audio(hdmi);
 		hdmi_wq_set_output(hdmi, hdmi->mute);
@@ -242,10 +239,7 @@ static void hdmi_wq_remove(struct hdmi *hdmi)
 	if (hdmi->ops->remove)
 		hdmi->ops->remove(hdmi);
 	#ifdef CONFIG_SWITCH
-	if ((hdmi->edid.baseaudio_support &&
-	     hdmi->edid.sink_hdmi) ||
-	     rk_fb_get_display_policy() == DISPLAY_POLICY_BOX)
-		switch_set_state(&(hdmi->switchdev), 0);
+	switch_set_state(&(hdmi->switchdev), 0);
 	#endif
 	list_for_each_safe(pos, n, &hdmi->edid.modelist) {
 		list_del(pos);
-- 
2.35.3

